File: scenarios/python-one-dark-pro/input.py
from dataclasses import dataclass


@dataclass
class Point:
    x: float
    y: float

    def distance(self, other: "Point") -> float:
        return ((self.x - other.x) ** 2 + (self.y - other.y) ** 2) ** 0.5


def main() -> None:
    points = [Point(i, i * 2) for i in range(5)]
    total = sum(p.distance(Point(0, 0)) for p in points)
    print(f"total distance = {total:.3f}")


if __name__ == "__main__":
    main()

File: scenarios/kotlin-ayu-mirage/input.kt
package com.example.app

import kotlin.math.sqrt

data class Point(val x: Double, val y: Double) {
    fun distanceTo(other: Point): Double {
        val dx = x - other.x
        val dy = y - other.y
        return sqrt(dx * dx + dy * dy)
    }
}

fun main() {
    val points = listOf(Point(0.0, 0.0), Point(3.0, 4.0))
    val total = points.windowed(2).sumOf { (a, b) -> a.distanceTo(b) }
    val label = "total distance = $total"
    println(label)
    points.forEachIndexed { i, p -> println("#$i -> (${p.x}, ${p.y})") }
}
